### ISSUE TITLE
Road to v1

### ISSUE THREAD

**Issue by martinnormark** [contextio/contextio-swift#2]
Hey @istx25 

I had a chat with @tonyblank over email about who was taking the lead on getting an updated iOS client library shipped and he mentioned you :+1: 

I was wondering if you could share your plans/roadmap or if there's any way you need help getting to a workable implementation of the API to start using it?

That said, for v1 in open source it's sometimes better to just stay focused on your own and getting it done before turning to the community for continuous iterations.

**Comment by tonyblank** [comment]
Just wanted to chime in with a :+1: here! @istx25, if it makes sense to you, feel free to outline the lib with a bunch of TODOs and @martinnormark and I can assist!

**Comment by istx25** [comment]
Hallo @tonyblank and @martinnormark!

My apologies for not responding as quick as I should've. A lot going on in my own startup these days. I'd love to contribute to the development of the Swift based API (and demonstration project) for ContextIO. I'll work on curating a list of things that need to be accomplished and we can all work away on it.

I think that the first and most important part would be re-engineering what was made in the [Objective-C](https://github.com/contextio/contextio-ios) version, and then add some awesome enhancements, improvements and new features. I'd love to contribute as much as my professional work will allow me to. I'll create a PR #3 for the new dependencies and classes I created over the last few days. There's not a lot of changes, but a few good additives. 

I would rather do this as a collaborative effort than doing the first version alone, so maybe we could organise tasks somewhere? I personally am just getting into [AlamoFire](http://www.github.com/AlamoFire/AlamoFire). We're using it in my product startup, Moments and I don't have a lot of experience with the API's (something to learn! :wink:)

**Comment by martinnormark** [comment]
Hey, no worries.

Since the API is quite extensive and the Objective-C client is rather big, it might be worth it to focus on getting an implementation of the [Lite API](http://context.io/docs/lite) up and running - just to stay focused on the basics first?

I've used Alamofire to some extend and it's a nice API, particularly the use of [Enums and URLRequestConvertible](https://github.com/Alamofire/Alamofire#crud--authorization) is nice, without being too clever at the cost of readability.

**Comment by tonyblank** [comment]
I agree that it makes sense to focus on Lite first to get coverage. It'd be
easier to fork the lite lib, change a few endpoints, and add in the methods
needed for the other 2.0 calls to get 100% coverage of 2.0.

On Tue, Feb 10, 2015 at 2:31 PM, Martin H. Normark <notifications@github.com

**Comment by istx25** [comment]
Woah! I didn't even know there was a _Lite API_. Definitely. If there is a simpler version, I agree that it's best to make the building blocks before the building itself. Would you be able to outline what specific functionality needs to be in the Lite API, @martinnormark? 

If you would be able to write the `URLRequestConvertible` enum, @martinnormark? One of my engineers working on Moments created ours in its own swift class. I think that we should keep that up. I think that it would be best if one of you wrote the initial to-do list because you know what your existing lite and 2.0 API's are capable of thoroughly. 

Looking forward to getting this show on the road.

**Comment by istx25** [comment]
Also: Could I be added to the repository as a contributor? It's better than having to work through a forked copy. More effective.

**Comment by istx25** [comment]
Thank you @tonyblank.

**Comment by tonyblank** [comment]
I've made a new team with access to the swift repo with write access - I've
invited both of you.

The Lite API is (more or less) a subset of the functionality of 2.0. The
2.0 API relies upon an index of email headers to power slightly more
advanced features that not all apps would need. The Lite API is simply a
translation / abstraction layer and while it has less functionality of 2.0,
it still can satisfy most use cases.

**Comment by TarangKhanna** [comment]
Hi,
I wanted to use the Context.io with swift for an app idea and found this. As I see it the code does not have much right now. Are you guys planning to use the REST API to bridge the swift lib with context.io api?

**Comment by tonyblank** [comment]
One of our community members, @istx25 <https://github.com/istx25>, started
this lib and @martinnormark <https://github.com/martinnormark> has been
helping as well.

**Comment by TarangKhanna** [comment]
Hmm, ok so I will just ask my question here, hopefully they see it.
So i tried this to get messages that was sent to my account from another email address:
 Alamofire.request(.GET, "https://api.context.io/2.0/accounts/myidhere/contacts/andanemailhere/messages", parameters: ["limit":"100" ])
                .response { (request, response, data, error) in
                    println(request)
                    println(response)
                    println(data)
                    println(error)
            }


But I only got this JSON:
<NSMutableURLRequest: 0x7f8e74063bc0> { URL: https://api.context.io/2.0/accounts/id/contacts/emailhttps://api.context.io/2.0/accounts/myIdHereAgain/contacts/anotherEmail/messages?limit=100 } { status code: 401, headers {
    Connection = "keep-alive";
    "Content-Encoding" = gzip;
    "Content-Length" = 73;
    "Content-Type" = "application/json";
    Date = "Sat, 07 Mar 2015 18:54:51 GMT";
    Server = Apache;
    Vary = "Accept-Encoding";
    "X-Powered-By" = "PHP/5.3.10-1ubuntu3.6";
} })
Optional(<7b227479 7065223a 22657272 6f72222c 22636f64 65223a31 30352c22 76616c75 65223a22 496e7661 6c696420 636f6e73 756d6572 206b6579 227d>)
nil

**Comment by istx25** [comment]
I think that it would be best if your question was in a new issue. This issue is for planning.

**Comment by TarangKhanna** [comment]
Ok will do

**Comment by tonyblank** [comment]
You'll have to sign the request with your key & secret. Context.IO supports
Oauth1 signed requests. Since I'm not familiar with swift, unsure the
correct lib to use. You can definitely duplicate the request you're trying
to make in our interactive console and take a look at the request params to
see how it should be formed.

Cheers,
Tony
